gui/src/ais.cpp

```diff
@@ -355,6 +355,24 @@ void AISDrawAreaNotices(ocpnDC &dc, ViewPort &vp, ChartCanvas *cp) {
 
                 break;
               }
+              case AIS8_001_22_SHAPE_SECTOR: {
+                wxPoint target_point;
+                double lat, lon;
+                double lat1 = sa->latitude;
+                double lon1 = sa->longitude;
+                GetCanvasPointPix(vp, cp, lat1, lon1, &target_point);
+                points.push_back(target_point);
+                ll_gc_ll(lat1, lon1, sa->left_bound_deg, sa->radius_m / 1852.0,
+                         &lat, &lon);
+                GetCanvasPointPix(vp, cp, lat, lon, &target_point);
+                points.push_back(target_point);
+                ll_gc_ll(lat1, lon1, sa->right_bound_deg, sa->radius_m / 1852.0,
+                         &lat, &lon);
+                GetCanvasPointPix(vp, cp, lat, lon, &target_point);
+                points.push_back(target_point);
+                dc.DrawPolygon(points.size(), &points.front());
```

Comment:
I think `draw_polygon = true;` instead of actually drawing the polygon belongs here